src/masternode-budget.cpp

```diff
@@ -81,26 +82,34 @@ bool IsBudgetCollateralValid(const uint256& nTxCollateralHash, const uint256& nE
         - nTime is never validated via the hashing mechanism and comes from a full-validated source (the blockchain)
     */
 
-    int conf = GetIXConfirmations(nTxCollateralHash);
+    nConf = GetIXConfirmations(nTxCollateralHash);
     if (!nBlockHash.IsNull()) {
         BlockMap::iterator mi = mapBlockIndex.find(nBlockHash);
         if (mi != mapBlockIndex.end() && (*mi).second) {
             CBlockIndex* pindex = (*mi).second;
             if (chainActive.Contains(pindex)) {
-                conf += chainActive.Height() - pindex->nHeight + 1;
+                nConf += chainActive.Height() - pindex->nHeight + 1;
                 nTime = pindex->nTime;
             }
         }
     }
 
-    nConf = conf;
+    return true;
+}
+
+bool IsBudgetCollateralValid(const uint256& nTxCollateralHash, const uint256& nExpectedHash, std::string& strError, int64_t& nTime, int& nConf, bool fBudgetFinalization)
+{
+    // Static checks. !TODO: move from here - they should be done only once (or after reorgs).
+    uint256 nBlockHash;
+    if (!CheckCollateral(nTxCollateralHash, nExpectedHash, strError, nTime, nConf, fBudgetFinalization))
```

Comment:
As is now, `nBlockHash` is not used, guess that you wanted to return `nBlockHash` from `CheckCollateral` and return false if it wasn't found.

Reply:
Yes, this is cleaned in a successive commit (ad1e9a291c55b5576bf9fa7ea463d29d7d534858).